### PR TITLE
Fixes unit tests for group search

### PR DESCRIPTION
**Pull request by svmudkavi**

### CODE REVIEW

File: spec/unit/connection_spec.rb [hunk 1]
```diff
@@ -14,29 +14,60 @@
 
   describe 'When verifying user groups' do
     before :all do
-      default_devise_settings!
-      reset_ldap_server!
+      #default_devise_settings!
```
Comment (justinIs) [inline spec/unit/connection_spec.rb]: do we need this commented out? can remove the code if it's not necessary

File: spec/unit/connection_spec.rb [hunk 2]
```diff
@@ -14,29 +14,60 @@
...
-        ::Devise.ldap_auth_username_builder = Proc.new() {|attribute, login, ldap| "#{login}"}
-        connection = Devise::LDAP::Connection.new(:login => admin.email, :password => admin.password, :admin => true)
-        assert_equal true, connection.in_group?('cn=admins,ou=testbase,dc=test,dc=com')
+        ::Devise.ldap_auth_username_builder = Proc.new() {|attribute, login, ldap|
+          "#{login}"
+        }
+
+        def mockResponse(group_lookup_attribute, user_lookup_attribute, group_name)
+          filter = generate_filter(group_lookup_attribute, user_lookup_attribute, group_name)
+          if(filter.to_s.byteslice(1..-1) == "&(#{user_lookup_attribute}=cn=example.admin@test.com,ou=people,dc=test,dc=com)(#{group_lookup_attribute}=cn=admins,ou=groups,dc=test,dc=com))".encode('UTF-8'))
```
Comment (justinIs) [inline spec/unit/connection_spec.rb]: is this the hack we were talking about? Let's comment on hacks with a TODO: this is a hack for x, needs a better way. type thing

Comment (svmudkavi) [inline spec/unit/connection_spec.rb]: Yep, this is the one. I'll add the comments.

File: spec/unit/connection_spec.rb (hunk 1 above)
Comment (svmudkavi) [inline spec/unit/connection_spec.rb]: Nope, we'll get rid of it

**Comment by justinIs** [comment]
👍

**Pull request merged**